curp/src/client_new/mod.rs

```diff
@@ -0,0 +1,168 @@
+#[derive(Debug, Clone)]
+pub struct ClientBuilder {
+    /// local server id
+    local_server_id: Option<ServerId>,
+    /// initial cluster version
+    cluster_version: Option<u64>,
+    /// client configuration
+    config: ClientConfig,
+    /// initial all members
+    all_members: Option<HashMap<ServerId, Vec<String>>>,
+}
+
+impl ClientBuilder {
+    /// Create a client builder
+    #[inline]
+    #[must_use]
+    pub fn new(config: ClientConfig) -> Self {
+        Self {
+            local_server_id: None,
+            config,
+            all_members: None,
+            cluster_version: None,
+        }
+    }
+
+    /// Set the local server id
+    #[inline]
+    #[must_use]
+    pub fn local_server_id(&mut self, id: ServerId) -> &mut Self {
+        self.local_server_id = Some(id);
+        self
+    }
+
+    /// Set the initial cluster version
+    #[inline]
+    #[must_use]
+    pub fn cluster_version(&mut self, cluster_version: u64) -> &mut Self {
+        self.cluster_version = Some(cluster_version);
+        self
+    }
+
+    /// Fetch initial all members from some endpoints if you do not know the whole members
+    ///
+    /// # Errors
+    ///
+    /// Return `tonic::Status` for connection failure or some server errors.
+    #[inline]
+    pub async fn fetch_all_members(
+        &mut self,
+        addrs: Vec<String>,
+    ) -> Result<&mut Self, tonic::Status> {
+        let propose_timeout = *self.config.propose_timeout();
+        let mut futs: FuturesUnordered<_> = addrs
+            .into_iter()
+            .map(|mut addr| {
+                if !addr.starts_with("http://") {
+                    addr.insert_str(0, "http://");
+                }
+                async move {
+                    let mut protocol_client = ProtocolClient::connect(addr).await.map_err(|e| {
+                        tonic::Status::cancelled(format!("cannot connect to addr, error: {e}"))
+                    })?;
+                    let mut req = tonic::Request::new(FetchClusterRequest::default());
+                    req.set_timeout(propose_timeout);
+                    let fetch_cluster_res = protocol_client.fetch_cluster(req).await?.into_inner();
+                    Ok::<FetchClusterResponse, tonic::Status>(fetch_cluster_res)
+                }
+            })
+            .collect();
+        let mut err = tonic::Status::invalid_argument("addrs is empty");
+        while let Some(r) = futs.next().await {
+            match r {
+                Ok(r) => {
+                    self.cluster_version = Some(r.cluster_version);
+                    self.all_members = Some(r.into_members_addrs());
+                    return Ok(self);
+                }
+                Err(e) => err = e,
+            }
+        }
+        Err(err)
+    }
+
+    /// Set the initial all members
+    #[inline]
+    #[must_use]
+    pub fn set_all_members(&mut self, all_members: HashMap<ServerId, Vec<String>>) -> &mut Self {
```

Comment:
Please modify the method name to `all_members` to unify the method naming style with other methods, like `local_server_id`.

Reply:
It was refactored in #582